### PR TITLE
[idyntree] Add new port

### PR DESCRIPTION
**Pull request by traversaro**
Add a port for the [iDynTree library](https://github.com/robotology/idyntree).

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/idyntree/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,51 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO robotology/idyntree
+    REF 51d5c740449b7daa6febb7b6268433366309ac2a # 8.1.0
```
Comment (Adela0814) [inline ports/idyntree/portfile.cmake:4]:
```suggestion
    REF “v${VERSION}”

```

**Review by Adela0814 — CHANGES_REQUESTED** [review]

**Comment by Adela0814** [comment]
@traversaro  I encountered an error while testing usage:
```
CMake Error at CMakeTest/CMakeLists.txt:13 (target_link_libraries):
  Target "CMakeTest" links to:

    iDynTree::idyntree-high-leveliDynTree::idyntree-estimation
```

**Comment by traversaro** [comment]
> @traversaro I encountered an error while testing usage:
> 
> ```
> CMake Error at CMakeTest/CMakeLists.txt:13 (target_link_libraries):
>   Target "CMakeTest" links to:
> 
>     iDynTree::idyntree-high-leveliDynTree::idyntree-estimation
> ```

Thanks, there was a typo, I fixed this in https://github.com/microsoft/vcpkg/pull/30810/commits/97574753bbe660285da0c82170c9109a826597e8 .

**Review by Adela0814 — DISMISSED** [review]

File: ports/idyntree/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS
+        ${FEATURE_OPTIONS}
+        -DIDYNTREE_USES_IPOPT:BOOL=OFF
+        -DIDYNTREE_USES_OSQPEIGEN:BOOL=OFF
+        -DIDYNTREE_USES_MATLAB:BOOL=OFF
+        -DIDYNTREE_USES_PYTHON:BOOL=OFF
+        -DIDYNTREE_USES_OCTAVE:BOOL=OFF
+        -DIDYNTREE_USES_LUA:BOOL=OFF
+        -DIDYNTREE_COMPILE_TESTS=OFF
```
Comment (BillyONeal) [inline ports/idyntree/portfile.cmake:25]: What about:

* YARP: https://github.com/robotology/idyntree/blob/e7bfd89372ddeb4a9930f067dc26386ac75cac1e/cmake/iDynTreeDependencies.cmake#L69
* qt5: https://github.com/robotology/idyntree/blob/e7bfd89372ddeb4a9930f067dc26386ac75cac1e/cmake/iDynTreeDependencies.cmake#L85
* arglib: https://github.com/robotology/idyntree/blob/e7bfd89372ddeb4a9930f067dc26386ac75cac1e/cmake/iDynTreeDependencies.cmake#L87
* WORHP https://github.com/robotology/idyntree/blob/e7bfd89372ddeb4a9930f067dc26386ac75cac1e/cmake/iDynTreeDependencies.cmake#L94

?

Comment (traversaro) [reply ports/idyntree/portfile.cmake:25]: Good point, I was not considering them as qt5 is actually not used upstream and should be removed, YARP is not used if IDYNTREE_COMPILE_TESTS is OFF, while arglib and WORHP are already set by default to OFF. Anyhow, just to play safe and set the options explicitly to OFF, thanks!

Comment (traversaro) [reply ports/idyntree/portfile.cmake:30]: Done in https://github.com/microsoft/vcpkg/pull/30810/commits/6726bac3c402737dbcd4ba0cfbddbd9cbccc74cb .

**Comment by BillyONeal** [comment]
Thanks!

**Comment by traversaro** [comment]
Thanks for merging!

**Pull request merged**